addons/bus/static/src/services/bus_service.js

```diff
@@ -89,7 +89,7 @@ export const busService = {
             worker.addEventListener('message', handleMessage);
         }
         send('update_last_notification_id', multiTab.getSharedValue('last_notification_id', 0));
-        browser.addEventListener('unload', () => send('leave'));
+        browser.addEventListener('pagehide', () => send('leave'));
```

Comment:
Hello @qsm-odoo, I just tested it, every works as expected. I'm only concerned about one thing: what if the page goes to the `frozen` state after the `pagehide` event ? Then the page switches back to the `active/hidden/passive` state and would be disconnected from the worker. If that is true, I think we should add another listener for the `resume` event in order to re-subscribe to the worker.

Reply:
My knowledge is very limited here, but from what I understand, we can check the `persisted` property of the `pagehide` event that is fired to know if a `freeze` or an `unload` should follow? https://developer.chrome.com/blog/page-lifecycle-api/#event-pagehide I guess it would be then be exactly like it is right now (still connected to the worker during the "frozen" state?).

I guess you'll know better what's the best solution between those two (disconnect from the worker during frozen state and reconnect after like you suggested or not disconnect from the worker during frozen state (like before)). I guess this is kinda the same. Would you push-force this PR yourself with the one you decide after testing? Or I can do it myself but it would probably be faster for you rather than me trying to understand the `bus` module and how to test it then asking you to test it anyway :grimacing: You tell me :)

Reply:
Sure, I'll take care of it as soon as possible :)

Reply:
Thanks @tsm-odoo, maybe you could take a look at this at the same time: maybe we should not ever show the disconnect notification while on the frontend (I am not sure if I ever saw it before websocket):
- Visitors of a website should never see this (or could it make sense?)
- When you are on the website app, now that you have an iframe of the website while being in the backend, you have that notification twice: once inside the iframe and once outside of it

@rdeodoo reported it a while ago and at the time I thought it was not a big deal because it only happens if the server is shut down once someone is on the page... but this seems to appear at other times, for example I cannot explain why it shows up on the screenshot of a pagespeed test (well I guess they close connexion to the runbot server once the website is loaded on their side but... still... strange):
![image](https://user-images.githubusercontent.com/10338094/191984825-18d5b6aa-1c33-45e9-8191-2f14d4ae7c2b.png)

Reply:
Note that I created a task internally back then: https://www.odoo.com/web#id=2922959&menu_id=4720&cids=1&action=4043&model=project.task&view_type=form

@bso-odoo actually already made a fix for this: https://github.com/odoo/odoo/pull/100173 :zap: 

So I guess that part of the discussion can be ignored here and move to that PR